### PR TITLE
Develop (v1)

### PR DESCRIPTION
**Pull request by axellorreyne**

### CODE REVIEW

File: src/popup/popup.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
+function main() {
+
+    const $loginbutton = document.getElementById('generate-button');
```
Comment (pheyvaer) [inline src/popup/popup.js:3]: I would rename the id to `login-button` to make its function more clear.

File: src/js/background.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+/**
+ * Rewrite request headers with correct credentials so that successful authentication may be achieved
+ * @param {Object} details - Request details passed on from blocking web request listener
+ * @returns {Object} - Object containing altered request headers, to be handled by the web request listener callback
+ */
+async function rewriteRequestHeaders(details) {
+
+    // TODO: find a more elegant way to catch the access token creation request called from getAccessToken()
+    if (details.method === "POST") {
```
Comment (pheyvaer) [inline src/js/background.js:48]: 1) Can you make an issue for this TODO?
2) Can you add to the README that only GET requests are supported. I assume that all POST requests will fail, right?

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+const CopyWebpackPlugin = require('copy-webpack-plugin')
+const NodePolyfillPlugin = require("node-polyfill-webpack-plugin");
+const path = require('path')
+//const webpack = require('webpack')
```
Comment (pheyvaer) [inline webpack.config.js:4]: Remove unused `require`.

**Review by pheyvaer — CHANGES_REQUESTED** [review]

File: src/popup/popup.js (hunk 1 above)
Comment (axellorreyne) [reply src/popup/popup.js]: This was renamed in 5d604c5f83ec276107c7331b4c57c03e12f1a4a9

File: src/js/background.js (hunk 2 above)
Comment (axellorreyne) [reply src/js/background.js:48]: Posts requests from the user will not fail or get blocked, they will just not have any authentication, if that is what you meant by 'fail'

I made an issue for this: #14 
Documentation for this was added to the readme in 5d604c5f83ec276107c7331b4c57c03e12f1a4a9

File: webpack.config.js (hunk 3 above)
Comment (axellorreyne) [reply webpack.config.js]: This was removed in 5d604c5f83ec276107c7331b4c57c03e12f1a4a9

**Review by pheyvaer — APPROVED** [review]

**Pull request merged**